app/javascript/components/SweetPhoneInput/SweetPhoneInput.js

```diff
@@ -73,10 +73,20 @@ class SweetPhoneInput extends Component {
   }
 
   onPhoneNumberChange = (value: string = '') => {
-    this.setState(prevState => ({
-      ...prevState,
-      phoneNumber: format(value, this.getCountryCode(), 'National'),
-    }));
+    this.setState(prevState => {
+      let newPhoneNumber;
+      if (value.length <= prevState.phoneNumber.length) {
+        newPhoneNumber = value;
+      } else {
+        newPhoneNumber = new asYouType(this.getCountryCode()).input(value);
+      }
+
+      return {
+        ...prevState,
+        prevPhoneNumber: prevState.phoneNumber,
```

Comment:
don't seem to need `prevPhoneNumber` since you'll be using `prevState.phoneNumber`.